ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -374,7 +374,12 @@ tryCollectFile hasFS@HasFS{..} env slot st@InternalState{..} (fileId, fileInfo)
               }
         | isCurrentNew  -> return st
         | otherwise     -> do
-            -- TODO: Rething if we want to reopen after
+            -- We reach this case if we have to garbage collect the current file
+            -- we are appending blocks to. I'm not sure how realistic this case
+            -- is and maybe we want to skip this garbage collection.
+            --
+            -- 'ReOpenFile' techinally truncates the file to 0 offset, so any
```

Comment:
`s/ReOpenFile/reOpenFile/` `s/techinally/technically/`

Ah, I now understand this case.

> I'm not sure how realistic this case is and maybe we want to skip this garbage collection.

Can you verify that this case is actually triggered in the tests? You can simply let it call `error` and see whether it is thrown. Maybe you'll need to play with the test parameters.

> I'm not sure how realistic this case is and maybe we want to skip this garbage collection.

I would replace this with an explanation:

> For this to happen, a garbage collection would have to be triggered for a slot ...

Reply:
> Can you verify that this case is actually triggered in the tests? You can simply let it call error and see whether it is thrown. Maybe you'll need to play with the test parameters.

@kderme Have you seen this happen?

Reply:
Yes, q-s-m triggers this case. I left the comment:
```
-- We reach this case if we have to garbage collect the current file 
-- we are appending blocks to. For this to happen, a garbage
-- collection would have to be triggered for a slot which is bigger
-- than any recently inserted blocks.
```